TypeCobol/Compiler/Scanner/TokensLine.cs

```diff
@@ -151,9 +158,9 @@ internal void RemoveDiagnosticsForToken(Token token)
         {
             if (token != null)
             {
-                foreach (Diagnostic diag in GetDiagnosticsForToken(token).ToArray())
+                foreach (Diagnostic diag in _ScannerDiagnostics.Where(diag => diag is TokenDiagnostic && ((TokenDiagnostic)diag).Token == token).ToArray())
```

Comment:
Why not keep GetDiagnosticsForToken(token).ToArray() ?

Reply:
Yes, you're right, I'll change that.